editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -108,168 +125,101 @@ private void configureNature(IProject project, MavenProject m2Project, IProgress
-			addFuseNature(project, monitor);
+			ProjectUtilities.addNatureToProject(project, RiderProjectNature.NATURE_ID);
 		}
 	}
 	
-	private void configureInternal(MavenProject mavenProject, IProject project, IProgressMonitor monitor)
+	private void configureFacet(MavenProject mavenProject, IProject project, IProgressMonitor monitor)
 			throws CoreException {
 
 		if (!isCamelConfigurable(mavenProject, project)) {
 			return;
 		}
 		IFacetedProject fproj = ProjectFacetsManager.create(project);
-
+		
 		if (fproj == null) {
 			// Add the modulecore nature
 			WtpUtils.addNatures(project);
 			ProjectUtilities.addNatureToProject(project, FacetedProjectNature.NATURE_ID);
 			fproj = ProjectFacetsManager.create(project);
-
 		}
+		
 		if (fproj != null) {
-			String camelVersion = getCamelVersion(project, mavenProject);
-			if (camelVersion != null) {
-				installDefaultFacets(fproj, camelVersion, monitor);
-			}
+			installDefaultFacets(project, mavenProject, fproj, monitor);
 		}
 	}
 
+	private void installDefaultFacets(IProject project, MavenProject mavenProject, IFacetedProject fproj, IProgressMonitor monitor)
+			throws CoreException {
+		String camelVersion = getCamelVersion(project, mavenProject);
+		if (camelVersion != null) {
+			IFacetedProjectWorkingCopy fpwc = fproj.createWorkingCopy();
+			IDataModel javaModel = DataModelFactory.createDataModel(new JavaFacetInstallDataModelProvider());
+			installFacet(fproj, fpwc, javaFacet, javaFacet.getLatestVersion(), javaModel, monitor);
+			installFacet(fproj, fpwc, m2eFacet, m2eFacet.getLatestVersion(), null, monitor);
+			IDataModel utilityModel = (IDataModel) new UtilityFacetInstallDataModelProvider().create();
+			installFacet(fproj, fpwc, utilFacet, utilFacet.getLatestVersion(), utilityModel, monitor);
+			installCamelFacet(fproj, fpwc, camelVersion, monitor);
+			fpwc.commitChanges(monitor);
+			configureNature(project, mavenProject, monitor); 
+		}
+	}
+	
 	private boolean isCamelConfigurable(MavenProject mavenProject, IProject project) {
 		// Look for a file that has parent "blueprint" and grandparent "OSGI-INF" or
 		// spring in folder META-INF
 		final Boolean[] found = new Boolean[1];
 		found[0] = false;
-		try {
-			project.accept(new IResourceVisitor() {
-				public boolean visit(IResource resource) throws CoreException {
-					if (resource.getName().endsWith(".xml") && resource.getParent().getName().equals("blueprint")
-							&& resource.getParent().getParent().getName().equals("OSGI-INF")) {
-						found[0] = true;
-					} else if (resource.getName().endsWith(".xml") && resource.getParent().getName().equals("spring")
-							&& resource.getParent().getParent().getName().equals("META-INF")) {
-						found[0] = true;
-					}					
-					return !found[0];
-				}
-			});
-		} catch (CoreException ce) {
 
+		// check for java dsl
+		IFile f = FuseIntegrationProjectCreatorRunnable.findJavaDSLRouteBuilderClass(project, new NullProgressMonitor());
```

Comment:
I think that the XML way is more frequent and takes less time. What do you think of inverting the 2 searches?
We also need to wait for the Build job to have finished.